src/main/java/me/botsko/prism/listeners/PrismBlockEvents.java

```diff
@@ -336,6 +338,49 @@ public void onLeavesDecay(final LeavesDecayEvent event) {
                 "Environment"));
     }
 
+    /**
+     * Primarily for tracking bed explosions in the nether and end.
+     * @param event BlockExplodeEvent
+     */
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+    public void onBlockExplode(BlockExplodeEvent event) {
+        if (!Prism.getIgnore().event("bed-explode", event.getBlock())) {
+            return;
+        }
+        //while it might be nice to check that its a bed - the block is already air
+        PlayerBed playerBed = weakCache.getIfPresent(event.getBlock().getLocation());
+        if (playerBed == null) {
+            return;
+        }
+        String source;
+        source = playerBed.player.getName();
+        List<Block> affected = event.blockList();
+        RecordingQueue.addToQueue(ActionFactory.createBlock("bed-explode", playerBed.bed, playerBed.player));
+        contructBlockEvent("bed-explode", source, affected);
+        weakCache.invalidate(event.getBlock().getLocation());
+    }
+
+    private final Cache<Location, PlayerBed> weakCache = CacheBuilder
```

Comment:
Codacy found an issue: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=6279951)